workflows/epitopeprediction.nf

```diff
@@ -94,43 +90,59 @@ import groovy.json.JsonSlurper
 
-    //
-    // SUBWORKFLOW: Read in samplesheet, validate and stage input files
-    //
-    INPUT_CHECK (
-        file(params.input)
-    )
-    ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
-    // TODO: OPTIONAL, you can use nf-validation plugin to create an input channel from the samplesheet with Channel.fromSamplesheet("input")
-    // See the documentation https://nextflow-io.github.io/nf-validation/samplesheets/fromSamplesheet/
-    // ! There is currently no tooling to help you write a sample sheet schema
-
-    INPUT_CHECK.out.meta
-                .branch {
-                    meta_data, input_file ->
-                        variant_compressed : meta_data.inputtype == 'variant_compressed'
-                            return [ meta_data, input_file ]
-                        variant_uncompressed :  meta_data.inputtype == 'variant'
-                            return [ meta_data, input_file ]
-                        peptide :  meta_data.inputtype == 'peptide'
-                            return [ meta_data, input_file ]
-                        protein :  meta_data.inputtype == 'protein'
-                            return [ meta_data, input_file ]
-                    }
-                .set { ch_samples_from_sheet }
+    // Function to read the alleles from a file or stage it from url
+    def readAlleles = { input ->
+        if (input.endsWith(".txt")) {
+            def file = file(input)
+            // Alleles are listed in the first line of the file
+            return file.readLines().get(0)
+        } else {
+            // Not a file path, return the original string
+            return input
+        }
+    }
+
+    ch_input = Channel.fromSamplesheet("input")
+    ch_input
+        .branch {
+            sample, alleles, mhc_class, filename ->
+                def allele_list = readAlleles(alleles)
+                validate_alleles(allele_list, mhc_class)
+                variant_compressed : filename.endsWith('.vcf.gz')
+                    return [[sample:sample, alleles:allele_list, mhc_class:mhc_class, inputtype:'variant_compressed'], filename ]
+                variant_uncompressed : filename.endsWith('.vcf') || filename.endsWith('.GSvar')
+                    return [[sample:sample, alleles:allele_list, mhc_class:mhc_class, inputtype:'variant'], filename ]
+                peptide : filename.endsWith('.tsv')
+                    return [[sample:sample, alleles:allele_list, mhc_class:mhc_class, inputtype:'peptide'], filename ]
+                protein : filename.endsWith('.fasta') || filename.endsWith('.fa')
+                    return [[sample:sample, alleles:allele_list, mhc_class:mhc_class, inputtype:'protein'], filename ]
+    } .set { ch_samples_from_sheet }
 
     // gunzip variant files
     GUNZIP_VCF (
         ch_samples_from_sheet.variant_compressed
     )
-    ch_versions = ch_versions.mix(GUNZIP_VCF.out.versions.ifEmpty(null))
+    ch_versions = ch_versions.mix(GUNZIP_VCF.out.versions)
```

Comment:
Are we sure that this is not needed anymore?

Reply:
Not needed anymore for versions channel. In fact will throw a warning with next tools update: https://github.com/nf-core/tools/issues/2506